IssueSyncTool/config/sync_config.json

```diff
@@ -0,0 +1,40 @@
+{
+   "source": ["github", "jira"],
+   "destination": ["rtc"],
+   "tracker": {
+      "github": {
+         "project" : "test-fullautomation",
+         "token": "<your_github_token>",
+         "repository": [
+            "python-issue-sync-tool",
+            "RobotFramework_AIO"
+         ],
+         "condition": {
+            "state": "open"
```

Comment:
Hi Ngoan,
is it also possible to exclude issues?
E.g. we can put a "nosync" tag to github?
If already synced such issues need also to be removed from RTC.
Or do you have an better idea. Somehow I want to have control what will be synced. This way each new issue would be synced by default. I'm not sure if this is good. We could also set a tag "new" by default, which has to be cleared manually for sync? Maybe this solved this. Then I could make a review before sync. Either I delete "new", or I set to "nosync" if it should be not synced...
Thank you,
 Thomas

Reply:
Hi Thomas,

>is it also possible to exclude issues?

Yes, you can add `exclude` below `condition` for exclude (negative) condition.
The below example (already in pdf documentation file) to exclude ticket without assignee and label `0.13.1`
```
{
   ...
   "tracker": {
      "github": {
         "project" : "test-fullautomation",
         "token": "<your-PAT>",
         "repository": [
            "python-jsonpreprocessor"
         ],
         "condition": {
            "state": "open",
            "exclude": {
               "assignee": "empty",
               "labels": "0.13.1"
            }
         }
      },
      ...
   }
   ... 
}
```

>If already synced such issues need also to be removed from RTC.

For the already sync issue, the original ticket title is updated to add rtc id (e.g [[ 2338592 ] An issue sync tool shall be developed](https://github.com/test-fullautomation/python-issue-sync-tool/issues/1)). 
Tool will know that issue is already created on RTC and only sync data (story point, version, status, description, ...) between original and synced tickets. 
Below table is my idea (already implemented partly) about attribute sync between original and rtc (mirrored) tickets
| Original tracker | Direction | Destination tracker  | Stage    |
|------------------|-----------|----------------------|----------|
| Title            | ->        | Title                | New      |
| URL              | ->        | Description          | New      |
| Assignee         | ->        | Assignee             | New      |
| Component        | ->        | Component            | New      |
| Status           | ->        | Status               | New/Sync |
| Title            | <-        | ID                   | New      |
| Story point      | <-        | Story point          | Sync     |
| Version          | <-        | Sprint/Version       | Sync     |

Stage `New` mean ticket is not existing on RTC: 
 - Create new rtc ticket
 - update original ticket title with rtc ID

 And stage `Sync` mean ticket is already existing
  - update rtc ticket with new status from original
  - update original with planning data (story point, version ) (--> This one is still under developped)

So I think `nosync` tag (label) is not required.

But we have to consider to update the condition for `Sync` stage because ticket from original may change state to `closed` (when completed by us) or we should define that ticket only change to closed state at end of sprint, ...

Please give your idea about them.

Thank you,
Ngoan

Reply:
Hi Ngoan,
sounds very good!
By default you should not sync when the assignee is empty.
This allows me to process all tickets before they are synced. It makes "new" tag not needed.

I think `nosync`  will be required if I don't want a ticket to be synced. There will be tasks like this.

With sprint and version I have problems. Sprint name will be different than AIO version. Maybe we create a tag with the sprint information in github/origin once synced. AIO versions won't follow sprints. This I don't want to be dependent on each other.

Closing will be done in github/origin only. There should be a sync back to RTC if closed.

Please give your ideas.

Thank you,
 Thomas

Reply:
Hi Thomas,

>This allows me to process all tickets before they are synced. It makes "new" tag not needed.

I've got it.

>Closing will be done in github/origin only. There should be a sync back to RTC if closed.

I have implemented it recent commits

>With sprint and version I have problems. Sprint name will be different than AIO version. Maybe we create a tag with the sprint information in github/origin once synced. AIO versions won't follow sprints. This I don't want to be dependent on each other.

This is what I am considering now. I aware about the different name between sprint on RTC planning vs our AIO version.
I think your proposal with tag (label) for sprint name (from RTC) is fine, it is available on github, gitlab and jira.
=> I will implement as this proposal.

Related to story point (please let me know if this attribute is not require for syncing): I also intent to use label (github and gitlab) when sync back  to original ticket, eg. `1-pst`, `2-pst`, ...

Please let me know if you see that any additional data (attribute) need to be added for syncing.

Thank you,
Ngoan

Reply:
Hi Ngoan,
story points should be provided in github/origin, too.
You have an idea where to put it?
Thank you,
 thomas

Reply:
Hi Thomas,

>story points should be provided in github/origin, too and be synced to RTC.

Oh, I assumed that story point is provided in RTC (during planning) and will be synced back to github, gitlab and jira.
=> I will update the direction to sync this attribute as your information. 

below is my ides for the story point
- On github and gitlab, we may create the labels which represents for the story point ( `1 point`, `2 points`, ... `8 points`). Then the issue can be add the appropriate label for the estimation.
- On Jira there is already `Original Estimate` field (which is counted by time eg. 3w 4d 12h)
I assume that 1 day will convert to 1 point, _**please let me know if my assumption is not correct**_

=> Tool will get these information from original tickets to create/update RTC workitem properly.

Please let me know whether you have any idea with my above proposal.

Thank you,
Ngoan

Reply:
Hi Ngoan,
I like the idea with the labels for the story points.
Usually a story point is not related to one day, but we start with this proposal. Anyhow there are not may issues in Jira.
Thank you,
 Thomas